src/test/tests/graphics/scenery/tests/unit/CatmullRomSplineTests.kt

```diff
@@ -16,29 +16,23 @@ import graphics.scenery.numerics.Random
 class CatmullRomSplineTests {
     private val logger by LazyLogger()
 
+    /**
+     * Tests if the curve object has actually the number of points defined in the class.
+     */
     @Test
     fun testCreation() {
         logger.info("This is the test for the CatmullRomSpline.")
-        val x1 = Random.randomFromRange(-30f, -10f)
-        val y1 = Random.randomFromRange(-30f, -10f)
-        val z1 = Random.randomFromRange(-30f, -10f)
-        val x2 = Random.randomFromRange(-10f, 20f)
-        val y2 = Random.randomFromRange(-10f, 20f)
-        val z2 = Random.randomFromRange(-10f, 20f)
-        val x3 = Random.randomFromRange(20f, 30f)
-        val y3 = Random.randomFromRange(20f, 30f)
-        val z3 = Random.randomFromRange(20f, 30f)
-        val x4 = Random.randomFromRange(30f, 100f)
-        val y4 = Random.randomFromRange(30f, 100f)
-        val z4 = Random.randomFromRange(30f, 100f)
+        val point1 = Random.randomVectorFromRange(3, -30f, -10f)
+        val point2 = Random.randomVectorFromRange(3, -9f, 20f)
+        val point3 = Random.randomVectorFromRange(3, 21f, 30f)
+        val point4 = Random.randomVectorFromRange(3, 31f, 100f)
 
-        val controlPoints = arrayListOf(GLVector(x1, y1, z1), GLVector(x2, y2, z2), GLVector(x3, y3, z3),
-        GLVector(x4, y4, z4))
+        val controlPoints = arrayListOf(point1, point2, point3, point4)
 
         val curve = CatmullRomSpline(controlPoints)
-
         assertNotNull(curve)
-        assertTrue(curve.catMullRomChain().size == 100)
+        //The computation of the Catmull Rom Spline can deliver sometimes one additional point
```

Comment:
Uh... sometimes? Could you please figure out why that is? The algorithm contains no random parts, so it should be entirely deterministic.

Reply:
My formulation was not good- the difference occurs because of the random vectors. It is a bit abstract, but in the end one gets 101 points if the distance between point 0 and point 1 is small relative to the difference between point 1 and point 2. In that case the while loop will spit out one more point.
However, it can't be more than 101 or less than 100. I will update the comment of course. What do you think, can it lead to problems?